client/src/components/addSourceWizard/addSourceWizardStepTwo.js

```diff
@@ -0,0 +1,462 @@
+import { SplitButton, MenuItem, Checkbox, Button, Icon, Form } from 'patternfly-react';
+import Store from '../../redux/store';
+import helpers from '../../common/helpers';
+import { getCredentials } from '../../redux/actions/credentialsActions';
+import { addSourceWizardField as FieldGroup } from './addSourceWizardField';
+import { apiTypes } from '../../constants';
+import { sourcesTypes, credentialsTypes } from '../../redux/constants';
+import _ from 'lodash';
+
+class AddSourceWizardStepTwo extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.initialState = {
+      source: {},
+      stepOneValid: false,
+
+      sourceType: '',
+      sourceName: '',
+      sourceNameError: null,
+      multiHostDisplay: '',
+      hosts: [],
+      hostsError: null,
+      credentials: [],
+      credentialsError: null,
+      port: '',
+      portError: null,
+      singleHostPort: '',
+      singleHostPortError: null,
+      satelliteVersion: '',
+      satelliteVersionError: null,
+      sslProtocol: '',
+      sslProtocolError: null,
+      sslCertVerify: '',
+      sslCertVerifyError: null,
+      disableSsl: '',
+      disableSslError: null,
+
+      allCredentials: []
+    };
+
+    this.state = { ...this.initialState, ...this.resetInitialState(props) };
+
+    helpers.bindMethods(this, [
+      'onChangeSourceName',
+      'onChangeCredential',
+      'onClickCredential',
+      'onChangeHost',
+      'onChangeHosts',
+      'onChangePort'
+    ]);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState(this.resetInitialState(nextProps));
+  }
+
+  componentDidMount() {
+    this.loadAllCredentials();
+  }
+
+  loadAllCredentials() {
+    this.props.getCredentials().then(response => {
+      this.setState(
+        {
+          allCredentials: response.value.data.results || []
+        },
+        () => this.updateCredentials()
+      );
+    });
+  }
+
+  updateCredentials(id, checked = true) {
+    const { credentials, allCredentials } = this.state;
+
+    if (id) {
+      let index = _.findIndex(allCredentials, { id: id });
+      if (index > -1) {
+        allCredentials[index].displayChecked = checked;
+      }
+    } else {
+      _.each(allCredentials, value => {
+        if (credentials.indexOf(value.id) > -1) {
+          value.displayChecked = true;
+        }
+      });
+    }
+
+    this.setState({
+      allCredentials: allCredentials
+    });
+  }
+
+  credentialInfo(id) {
+    const { allCredentials } = this.state;
+    return _.find(allCredentials, { id: id }) || {};
+  }
+
+  resetInitialState(nextProps) {
+    let credentials = nextProps.source[apiTypes.API_SOURCE_CREDENTIALS];
+    let remappedCredentials = credentials || [];
+
+    if (_.size(credentials) && credentials[0].id) {
+      remappedCredentials = credentials.map(val => val.id);
+    }
+
+    return {
+      source: nextProps.source || {},
+      sourceType: nextProps.source[apiTypes.API_SOURCE_TYPE] || '',
```

Comment:
Still not sure I'm a huge fan of this but I do see the value.

Reply:
yeah, i'm not super keen on it either, it gets clunky at a certain level. Next round i'm good with removing, or finding a better pattern